geopyspark/geotrellis/layer.py

```diff
@@ -1010,7 +1015,23 @@ def tile_to_layout(self, layout_definition, resample_method=ResampleMethod.NEARE
             :class:`~geopyspark.geotrellis.rdd.TiledRasterLayer`
         """
 
-        srdd = self.srdd.tileToLayout(layout_definition, ResampleMethod(resample_method))
+        resample_method = ResampleMethod(resample_method)
+
+        if isinstance(layout, LayoutDefinition):
+            srdd = self.srdd.tileToLayout(layout, resample_method)
+
+        elif isinstance(layout, Metadata):
+            srdd = self.srdd.tileToLayout(layout.layout_definition, resample_method)
```

Comment:
Needs to account for possible of mismatch in CRS between current RDD and metadata